server/server.js

```diff
@@ -1,15 +1,158 @@
+      "The Complete London 2012 Opening Ceremony | London 2012 Olympic Games",
+    url: "https://www.youtube.com/watch?v=4As0e4de-rI",
+    rating: 3211,
+  },
+  {
+    id: 283634,
+    title: "Learn Unity - Beginner's Game Development Course",
+    url: "https://www.youtube.com/watch?v=gB1F9G0JXOo",
+    rating: 211,
+  },
+  {
+    id: 562824,
+    title: "Cracking Enigma in 2021 - Computerphile",
+    url: "https://www.youtube.com/watch?v=RzWB5jL5RX0",
+    rating: 111,
+  },
+  {
+    id: 442452,
+    title: "Coding Adventure: Chess AI",
+    url: "https://www.youtube.com/watch?v=U4ogK0MIzqk",
+    rating: 671,
+  },
+  {
+    id: 536363,
+    title: "Coding Adventure: Ant and Slime Simulations",
+    url: "https://www.youtube.com/watch?v=X-iSQQgOd1A",
+    rating: 76,
+  },
+  {
+    id: 323445,
+    title: "Why the Tour de France is so brutal",
+    url: "https://www.youtube.com/watch?v=ZacOS8NBK6U",
+    rating: 73,
+  },
+];
 
 // GET "/"
-app.get("/", (req, res) => {
-  // Delete this line after you've confirmed your server is running
-  res.send({ express: "Your Backend Service is Running" });
+app.get("/videos", async (req, res) => {
+  try {
+    const query = `SELECT * FROM videos`;
+    const result = await db.query(query);
+    const videos = result.rows;
+    res.status(200).json(videos);
+  } catch (error) {
+    res.status(404).json({ error: "Fetch" });
+  }
+});
+
+//POST "/"
+app.post("/videos", (req, res) => {
+  //const { title, url } = req.body;
+  // const title = req.body.title;
+  // const url = req.body.url;
+  // if (!title || !url) {
+  //   return res.status(400).json({
+  //     result: "Failure",
+  //     message: "Video could not be saved",
+  //   });
+  // } else {
+  //   let nextVideoId = videos[videos.length - 1].id + 1;
+  //   const newAddedVideo = {
+  //     id: nextVideoId,
+  //     title: title,
+  //     url: url,
+  //   };
+  //   console.log(newAddedVideo);
+  //   videos.push(newAddedVideo);
+  //   res.status(201).json(videos);
+  // }
+  const title = req.body.title;
+  const url = req.body.url;
+  const rating = req.body.rating;
+  const query = `INSERT INTO videos (title, url, rating)
+VALUES ($1, $2, $3)`;
+  db.query(query, [title, url, rating])
+    .then(() => {
+      res.status(201).send("New video succesfully added!");
+    })
+    .catch((error) => {
+      console.log(error);
+    });
+});
+
+// GET "/{id}"
+app.get("/:id", (req, res) => {
+  const videoId = parseInt(req.params.id);
+  const video = videos.find((singleVideo) => singleVideo.id === videoId);
+  if (!video) {
+    return res.status(404).json({
+      result: "Failure",
+      message: "Video not found",
+    });
+  } else {
+    res.status(200).json(video);
+  }
+});
+
+// DELETE "/{id}"
+
+app.delete("/videos/:id", (req, res) => {
+  const videoId = parseInt(request.params.id);
+  const singleVideo = videos.find((singleVideo) => singleVideo.id === videoId);
+  if (singleVideo) {
+    videos.splice(singleVideo, 1);
```

Comment:
Splice takes as its first argument an _index_ not a value - are you sure this code works?